public/components/trace_analytics/components/traces/traces_custom_indices_table.tsx

```diff
@@ -47,218 +45,117 @@ interface TracesLandingTableProps {
   openTraceFlyout?: (traceId: string) => void;
   jaegerIndicesExist: boolean;
   dataPrepperIndicesExist: boolean;
+  tracesTableMode: TraceQueryMode;
+  setTracesTableMode: React.Dispatch<React.SetStateAction<TraceQueryMode>>;
 }
 
 export function TracesCustomIndicesTable(props: TracesLandingTableProps) {
   const { columnItems, items, refresh, mode, loading, getTraceViewUri, openTraceFlyout } = props;
   const [showAttributes, setShowAttributes] = useState(false);
+  const [isTitlePopoverOpen, setIsTitlePopoverOpen] = useState(false);
+  const [tableOptions, setTableOptions] = useState<EuiSelectableOption[]>(
+    TRACE_TABLE_OPTIONS.map((obj) =>
+      obj.key === props.tracesTableMode ? { ...obj, checked: 'on' } : obj
+    )
+  );
```

Comment:
```suggestion
  const [tableOptions, setTableOptions] = useState<EuiSelectableOption[]>(
    () => TRACE_TABLE_OPTIONS.map((obj) =>
      obj.key === props.tracesTableMode ? { ...obj, checked: 'on' } : obj
    )
  );
```
compute initial value once should be enough

Reply:
updated thanks to lazy init.